### PR TITLE
Add nuget generation for release/3.0

### PR DESCRIPTION
**Pull request by kbaladurin**

### CODE REVIEW

**Comment by kbaladurin** [comment]
@gbalykov @alpencolt PTAL

File: jobs/scripts/check_version.sh [hunk 1]
```diff
@@ -111,17 +114,32 @@ for list in ${versionlist[@]}; do
 done
 
 nupkg_name="${pkgname}.${fullversion}.nupkg"
-feedlist=( "https://www.nuget.org/api/v2/package"
+feedlistv2=( "https://www.nuget.org/api/v2/package"
```
Comment (gbalykov) [inline jobs/scripts/check_version.sh:117]: Why was this renamed?

File: jobs/generation/Utilities.groovy [hunk 2]
```diff
@@ -26,18 +26,47 @@ class Utilities {
...
+  def private static String getDockerImage(String branch) {
+    switch (branch) {
+      case 'release/2.0.0':
+        return 'hqueue/dotnetcore:ubuntu1404_cross_prereqs_v4-tizen_rootfs'
+      case 'release/2.1':
+      case 'release/2.2':
+        return 'tizendotnet/dotnet-buildtools-prereqs:ubuntu-16.04-cross-e435274-20180426002255-tizen-rootfs-5.0m1'
+      case 'release/3.0':
+      case 'master':
+        return 'tizendotnet/dotnet-buildtools-prereqs:ubuntu-16.04-cross-10fcdcf-20190208200917-tizen-rootfs-5.0m2'
```
Comment (gbalykov) [inline jobs/generation/Utilities.groovy:38]: Shouldn't this be 5.5m1?

File: jobs/generation/Utilities.groovy [hunk 3]
```diff
@@ -26,18 +26,47 @@ class Utilities {
...
+
+  def private static String getTizenVersion(String branch) {
+    switch (branch) {
+      case 'release/2.0.0':
+        return 'tizen.4.0.0'
+      case 'release/2.1':
+      case 'release/2.2':
+      case 'release/3.0':
+      case 'master':
+        return 'tizen.5.0.0'
```
Comment (gbalykov) [inline jobs/generation/Utilities.groovy:53]: Same as above, shouldn't it be tizen.5.5 for master and release/3.0?

File: jobs/generation/Utilities.groovy [hunk 4]
```diff
@@ -66,9 +95,13 @@ class Utilities {
               not {
                 stringsMatch("\${version}", "stable", false)
               }
+            }
```
Comment (gbalykov) [inline jobs/generation/Utilities.groovy:98]: This } is related to `and` above. Previous condition was `!stable && !servicing && !rtm`, why has this changed?

File: jobs/generation/Utilities.groovy [hunk 5]
```diff
@@ -140,27 +173,27 @@ class Utilities {
...
         } else if (project == 'corefx') {
-          if (branch == 'master') {
+          if (branch == 'master' || branch == 'release/3.0') {
             // Build command for CoreFX has changed: see https://github.com/dotnet/corefx/pull/32798/files and https://github.com/dotnet/corefx/commit/66392f577c7852092f668876822b6385bcafbd44
 
-            shell("${dockerCommand} ./build.sh -\${config} /p:ArchGroup=\${targetArch} /p:RuntimeOS=tizen.5.0.0 /p:PortableBuild=false ${passCI} \${buildIdOpts} \${stableOpts} \${packageOpts} /p:BinPlaceNETCoreAppPackage=true /p:OverridePackageSource=https:%2F%2Ftizen.myget.org/F/dotnet-core/api/v3/index.json ${authorsOpts}")
+            shell("${dockerCommand} ./build.sh --configuration \${config} /p:ArchGroup=\${targetArch} /p:RuntimeOS=${getTizenVersion(branch)} /p:PortableBuild=false /p:EnableNgenOptimization=false ${passCI} \${buildIdOpts} \${stableOpts} \${packageOpts} /p:BinPlaceNETCoreAppPackage=true /p:OverridePackageSource=https:%2F%2Ftizen.myget.org/F/dotnet-core/api/v3/index.json ${authorsOpts}")
```
Comment (gbalykov) [inline jobs/generation/Utilities.groovy:189]: Why is `EnableNgenOptimization` disabled?

File: jobs/generation/Utilities.groovy [hunk 6]
```diff
@@ -140,27 +173,27 @@ class Utilities {
...
+        if (project == 'corefx' && (branch == 'master' || branch == 'release/3.0')) {
           // Master uses Arcade for build and requires --ci option to be passed
           passCI = '--ci'
         }
 
         def authorsOpts = '/p:Authors=Tizen'
 
         if (project == 'coreclr') {
-          shell("${dockerCommand} ./build.sh cross \${config} \${targetArch} cmakeargs -DFEATURE_GDBJIT=TRUE stripSymbols -PortableBuild=false -- \${buildIdOpts} \${stableOpts} \${packageOpts} ${authorsOpts}")
+          shell("${dockerCommand} ./build.sh cross \${config} \${targetArch} cmakeargs -DFEATURE_GDBJIT=TRUE cmakeargs -DFEATURE_PREJIT=true stripSymbols -PortableBuild=false -- \${buildIdOpts} \${stableOpts} \${packageOpts} ${authorsOpts}")
```
Comment (gbalykov) [inline jobs/generation/Utilities.groovy]: I think we should add FEATURE_NGEN_RELOCS_OPTIMIZATIONS here too, so that nugets are consistent with rpms

File: jobs/scripts/check_version.sh (hunk 1 above)
Comment (kbaladurin) [reply jobs/scripts/check_version.sh:117]: It isn't necessary, I renamed it to distinguish from another `feedlist` that uses another web api.

File: jobs/generation/Utilities.groovy (hunk 2 above)
Comment (kbaladurin) [reply jobs/generation/Utilities.groovy:38]: Yes, but firstly we need to add support for `tizen.5.5.0` rid.

File: jobs/generation/Utilities.groovy (hunk 4 above)
Comment (kbaladurin) [reply jobs/generation/Utilities.groovy:98]: Previous code leads to generate just `!rtm`, now it corresponds to `!stable && !servicing && !rtm`

File: jobs/generation/Utilities.groovy (hunk 5 above)
Comment (kbaladurin) [reply jobs/generation/Utilities.groovy:189]: To use `EnableNgenOptimization` option in official build we need to [restore](https://github.com/dotnet/corefx/blob/2e6661d4cceeff2549b0299ec54e9f3b34c09b20/eng/pipelines/corefx-base.yml#L137) internal tools from authenticated channel, but we don't have credential for it.
```
$ ~/workspace/release/corefx_master_armel_Release/netcore-jenkins/dotnet-dev/dotnet restore --configfile eng/internal/NuGet.config --packages ./.packages eng/common/internal/Tools.csproj

Configuring...
--------------
A command is running to populate your local package cache to improve restore speed and enable offline access. This command takes up to one minute to complete and only runs once.
Decompressing 100% 6025 ms
Expanding 100% 5784 ms
  Restoring packages for /var/lib/jenkins/workspace/release/corefx_master_armel_Release/dotnet_corefx/eng/common/internal/Tools.csproj...
/var/lib/jenkins/workspace/release/corefx_master_armel_Release/netcore-jenkins/dotnet-dev/sdk/2.1.403/NuGet.targets(114,5): error : Unable to load the service index for source https://devdiv.pkgs.visualstudio.com/_packaging/dotnet-core-internal-tooling/nuget/v3/index.json. [/var/lib/jenkins/workspace/release/corefx_master_armel_Release/dotnet_corefx/eng/common/internal/Tools.csproj]
/var/lib/jenkins/workspace/release/corefx_master_armel_Release/netcore-jenkins/dotnet-dev/sdk/2.1.403/NuGet.targets(114,5): error :   Response status code does not indicate success: 401 (Unauthorized). [/var/lib/jenkins/workspace/release/corefx_master_armel_Release/dotnet_corefx/eng/common/internal/Tools.csproj]
```

**Review by gbalykov — APPROVED** [review]
LGTM

File: jobs/generation/Utilities.groovy (hunk 6 above)
Comment (kbaladurin) [reply jobs/generation/Utilities.groovy]: done
